Watchman.Web/Areas/JwtAuth/Infrastructure/JwtAuthManager.cs

```diff
@@ -0,0 +1,126 @@
+﻿using Microsoft.IdentityModel.Tokens;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Immutable;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Watchman.Web.Areas.JwtAuth.Infrastructure
+{
+    public class JwtAuthManager : IJwtAuthManager
+    {
+        public IImmutableDictionary<string, RefreshToken> UsersRefreshTokensReadOnlyDictionary => _usersRefreshTokens.ToImmutableDictionary();
+        private readonly ConcurrentDictionary<string, RefreshToken> _usersRefreshTokens;  // can store in a database or a distributed cache
+        private readonly JwtTokenConfig _jwtTokenConfig;
+        private readonly byte[] _secret;
+
+        public JwtAuthManager(JwtTokenConfig jwtTokenConfig)
+        {
+            _jwtTokenConfig = jwtTokenConfig;
+            _usersRefreshTokens = new ConcurrentDictionary<string, RefreshToken>();
+            _secret = Encoding.ASCII.GetBytes(jwtTokenConfig.Secret);
```

Comment:
jak używasz prywatnych pól to dodawaj na początku **this.**
wszędzie gdzie masz same nazwy pól to tam wszędzie dodaj te this.

Reply:
ok, poprawiam